data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py

```diff
@@ -0,0 +1,165 @@
+# Marcus Andersson - Percolator Project
+# Script that tests the performances of percolator for different data input sizes. Produces a graph image when finished.
+
+import os
+import sys
+import re
+
+from argparse import ArgumentParser
+import time
+import contextlib
+import subprocess
+from subprocess import Popen, PIPE
+from decimal import Decimal
+import pathlib
+import datetime
+from pathlib import Path
+import urllib
+from urllib.request import urlopen, Request
+from urllib.error import URLError, HTTPError
+import tarfile
+
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+success = True
+
+
+scriptDirectory = pathlib.Path(__file__).parent.absolute()
+testScriptPath = str(scriptDirectory) + "/" + "IntegrationTest_Percolator_Speed.py"
+testDataFolderPath = str(scriptDirectory) + "/" + "testData"
+testDataFilePath = ""
+
+path = Path(testDataFolderPath)
+path.mkdir(parents=False, exist_ok=True)
+
+def getArguments():
+    parser = ArgumentParser(description="Measure speed of perculator application with varying input and creates a graph.")
+    parser._action_groups.pop()
+    required = parser.add_argument_group('required arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    required.add_argument('-d','--data', type=str, metavar='', required=True, help="Path to input-data used by percolator.")
+    optional.add_argument('-py','--python_version', type=str, metavar='', default="python3" , required=False, help="Python version to call other python modules from.")
+    optional.add_argument('-r','--runs', type=int, metavar='', default=3, required=False, help="Number of tests to evaluate.")
+    optional.add_argument('-f','--flags', type=str, default = "" , metavar='', required=False, help="Flags to use with the executable.")
+    optional.add_argument('-a','--await_tests', default=False, action="store_true", required=False, help="Wait for others processes running this script to finish before starting.")
+    optional.add_argument('-c','--comments', type=str, default="", metavar='', required=False, help="Comments regarding test details.")
+    optional.add_argument('-fname','--filename', type=str, default="barFigure", metavar='', required=False, help="Name of produced charts.")
+    return parser.parse_args()
+
+def makeReducedFile(fileNameOriginal, fileNameReduced, numLines):
+    fileReduced = open(fileNameReduced, "w")
+    counter = 0
+    with open(fileNameOriginal) as f:
+        for line in f:
+            counter += 1
+            if (counter > numLines):
+                break
+            fileReduced.write(line)
+    fileReduced.close()
+
+from itertools import (takewhile,repeat)
+def rawincount(filename): ####https://stackoverflow.com/a/27518377/7202012
```

Comment:
Confusing function name -- rename to indicate what it's actually doing, since it's not at all obvious from reading the source.